tuf/ngclient/updater.py

```diff
@@ -0,0 +1,575 @@
+# Copyright 2020, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""TUF client workflow implementation.
+"""
+
+import fnmatch
+import logging
+import os
+from typing import Dict, Optional
+from urllib import parse
+
+from securesystemslib import exceptions as sslib_exceptions
+from securesystemslib import hash as sslib_hash
+from securesystemslib import util as sslib_util
+
+from tuf import exceptions
+from tuf.ngclient._internal import download, metadata_bundle, requests_fetcher
+from tuf.ngclient.fetcher import FetcherInterface
+
+# Globals
```

Comment:
Hmm, I feel these should be in a separate configuration class in a separate module, [like so](https://github.com/trishankatdatadog/tuf-on-a-plane/blob/ee8049a63ea439d72509f6e876eef08fb50c6ab1/src/tuf_on_a_plane/config.py). Otherwise, harder to make for safe concurrency.

Reply:
I fee like we have an issue or at least have planned _something_ here... @sechkova better memory than me?

Totally agreed on the reasoning

Reply:
Maybe https://github.com/theupdateframework/tuf/issues/1402? Also https://github.com/theupdateframework/tuf/issues/1273 for context.

Reply:
yes that's it, thank you